src/Compilers/VisualBasic/Portable/Semantics/TypeInference/TypeArgumentInference.vb

```diff
@@ -1630,20 +1634,30 @@ HandleAsAGeneralExpression:
                 parameterType As TypeSymbol,
                 param As ParameterSymbol,
                 digThroughToBasesAndImplements As MatchGenericArgumentToParameter,
-                inferenceRestrictions As RequiredConversion
+                inferenceRestrictions As RequiredConversion,
+                ByRef addedHints As Boolean
             ) As Boolean
 
+                If Not RefersToGenericParameterToInferArgumentFor(parameterType) Then
+                    Return True
+                End If
+
                 ' First try to the things directly. Only if this fails will we bother searching for things like List->IEnumerable.
-                Dim Inferred As Boolean = InferTypeArgumentsFromArgumentDirectly(
+                Dim addedDirectHints As Boolean = False
+                Dim directInferenceResult As Boolean = InferTypeArgumentsFromArgumentDirectly(
```

Comment:
Offline, @AlekseyTS explained to me that this is being used to fix a separate, unreported bug in the VB compiler whereby an invocation fails in certain situations when one argument fails to contribute to the inference of one type parameter, even though that type parameter can be inferred from another argument. See the change in TypeArgumentInference.vb for a test more focused on that bug. However, it seems that this fix should be done on a type parameter by type parameter basis, as the language requires *each* type parameter to be inferred from some argument (not just one of them). By tracking this as a single bool, it would appear to permit an invocation to succeed (because some type parameter was inferred from an argument) even though some *other* type parameter was *not* inferred from an argument. I think a correct fix would require a `Dictionary<TypeParameterSymbol, Boolean>` rather than a single `Boolean` to be correct.

I would suggest either fixing this correcly, or leaving this bug unfixed in VB for now (i.e. reverting the VB changes).